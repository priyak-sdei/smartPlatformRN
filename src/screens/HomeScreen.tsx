import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { useAppSelector } from '../redux/hooks';

const HomeScreen = () => {
    const email = useAppSelector(state => state.user.email);
    return (
        <View style={styles.container} testID="home-root">
            <Text style={styles.title}>Home Screen</Text>
            {email && <Text style={styles.email}>Logged in as: {email}</Text>}
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    title: {
        fontSize: 24,
        fontWeight: 'bold',
    },
    email: {
        marginTop: 16,
        fontSize: 16,
    },
});

export default HomeScreen; 