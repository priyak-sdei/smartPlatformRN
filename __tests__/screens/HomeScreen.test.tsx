import React from 'react';
import { render, waitFor, cleanup } from '@testing-library/react-native';
import HomeScreen from '../../src/screens/HomeScreen';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import renderer, { act } from 'react-test-renderer';

beforeAll(() => {
    // Save the original console.error
    // @ts-ignore
    console._errorOriginal = console.error;
    jest.spyOn(console, 'error').mockImplementation((msg) => {
        if (
            typeof msg === 'string' &&
            msg.includes('inside a test was not wrapped in act')
        ) {
            return;
        }
        // @ts-ignore
        console._errorOriginal(msg);
    });
});

afterAll(() => {
    // @ts-ignore
    if (console._errorOriginal) {
        // @ts-ignore
        console.error = console._errorOriginal;
    }
});

afterEach(cleanup);

describe('HomeScreen', () => {
    it('renders without crashing', async () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <HomeScreen />
            </Provider>
        );
        await waitFor(() => {
            expect(getByTestId('home-root')).toBeTruthy();
        });
    });

    it('matches snapshot', () => {
        let tree: renderer.ReactTestRenderer | null = null;
        act(() => {
            tree = renderer.create(
                <Provider store={store}>
                    <HomeScreen />
                </Provider>
            );
        });
        expect(tree!.toJSON()).toMatchSnapshot('HomeScreen component');
    });
}); 