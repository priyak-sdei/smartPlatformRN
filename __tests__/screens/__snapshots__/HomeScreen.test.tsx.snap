// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen matches snapshot: HomeScreen component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="home-root"
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Home Screen
  </Text>
</View>
`;
